### PR TITLE
Add Vagrantfile to project

### PR DESCRIPTION
**Pull request by lucasmoura**
This MR adds a Vagrantfile to the project, with the aim at helping automating the creation of a development environment for new users. 

Vagrant basically will create a virtual machine running Debian stretch, install all the necessary packages and run the unit tests to see if the environment is okay.

Currently, I am using virtualbox as the virtual machine provider, but this can be updated if necessary

### CODE REVIEW

**Comment by dcrankshaw** [comment]
jenkins test this please

File: README.md [hunk 1]
```diff
@@ -132,6 +132,33 @@ make
...
 ../bin/start_clipper.sh
+
+```
+
+__Using Vagrant:__
+
+You could use vagrant to automatically create a development environment. Vagrant
+will create a virtual machine containing this environment. To use
+that approach you need to install both vagrant and virtualbox. Once that is
+done, just do the following commands to initialize vagrant.
```
Comment (dcrankshaw) [inline README.md]: Some minor grammar edits:

"You can use Vagrant to automatically create a development environment in a virtual machine. This is one method for using Windows for Clipper development. Before you start, you must have [Vagrant](https://www.vagrantup.com/) and [VirtualBox](https://www.virtualbox.org/wiki/VirtualBox) installed. Once you have them installed, perform the following command to initialize Vagrant."

File: README.md [hunk 2]
```diff
@@ -132,6 +132,33 @@ make
...
+
+```sh
+vagrant up
+```
+
+And the environment will be created for you. You can now edit Clipper files
+in your local machine and when you want to test your changes, you can just
+perform the following commands:
```
Comment (dcrankshaw) [inline README.md]: change "in your local machine" to "on your local machine"

File: README.md [hunk 3]
```diff
@@ -132,6 +132,33 @@ make
...
+
+```sh
+# Logs in the virtual machine
```
Comment (dcrankshaw) [inline README.md]: "# Log in to the virtual machine"

File: README.md [hunk 4]
```diff
@@ -132,6 +132,33 @@ make
...
+vagrant ssh
+
+# Enter root dir
```
Comment (dcrankshaw) [inline README.md]: "# Enter root Clipper directory"

File: Vagrantfile [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+Vagrant.configure("2") do |config|
+  config.vm.box = "debian/testing64"
+
+  # Disable automatic box update checking. If you disable this, then
+  # boxes will only be checked for updates when the user runs
+  # `vagrant box outdated`. This is not recommended.
+  config.vm.box_check_update = false
```
Comment (dcrankshaw) [inline Vagrantfile]: What's the purpose of this setting? Is this standard practice? I haven't used Vagrant before.

**Review by dcrankshaw — CHANGES_REQUESTED** [review]
Hey this is very cool! I just have some minor edits for the documentation.

File: Vagrantfile (hunk 5 above)
Comment (lucasmoura) [reply Vagrantfile]: No, this is not standard practice. I was using that to just speed up the process of initializing vagrant, but I forgot to remove it :(. I will delete this line in my updated patch.

**Comment by lucasmoura** [comment]
@dcrankshaw Done, I have made the requested changes :)

**Review by dcrankshaw — APPROVED** [review]

**Comment by dcrankshaw** [comment]
jenkins test this please

**Comment by dcrankshaw** [comment]
LGTM

**Comment by dcrankshaw** [comment]
jenkins test this please

**Pull request merged**